### PR TITLE
Add better error validation message for username

### LINKED ISSUE #5829: Username with a space in it needs better validation error message
This is small but important - someone wrote in saying they got a "username invalid" message when trying to sign up; I tried, and realized that if you enter a space in a username, it says "invalid" but doesn't say why. 

Screen Shot 2019-06-03 at 6 42 29 PM

We should offer a validation error message specifying the formatting rules for usernames!

I think this should go here: 

https://github.com/publiclab/plots2/blob/faa5a12e86bf7944dca43134f649947f03ca96a6/app/models/user.rb#L60-L61

### PR DESCRIPTION
**Pull request by Tlazypanda**
Fixes #5829 

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

# Screenshot  
Screenshot from 2020-01-17 02-03-19

### CODE REVIEW

**Comment by SidharthBansal** [comment]
@VladimirMikulic kindly review

**Comment by Tlazypanda** [comment]
@jywarren @VladimirMikulic updated the pr and added ss of the change.Removed authlogic login for this case since the message was fixed in that case...also noticed that tests were failing because there was no check for length of username so added that check.Kindly review :v: Thanks :)

File: app/models/user.rb [hunk 1]
```diff
@@ -6,9 +6,6 @@ def validate(record)
   end
 end
 
-# Overwrites authlogic username regex to allow one character usernames
-Authlogic::Regex::LOGIN = /\A[A-Za-z\d_\-]*\z/
```
Comment (VladimirMikulic) [inline app/models/user.rb]: @jywarren is this line safe to remove?

Comment (jywarren) [reply app/models/user.rb]: I would say probably not - i think we need that? Can we look at the pull request that generated that line, using the "blame" view of that file on GitHub?

Thank you!

**Comment by Tlazypanda** [comment]
@jywarren @VladimirMikulic updated the pr :) Kindly review :v:

File: app/models/user.rb (hunk 1 above)
Comment (Tlazypanda) [reply app/models/user.rb]: @jywarren @VladimirMikulic Have updated the pr keeping in mind the changes.Thanks :)

**Comment by SidharthBansal** [comment]
:tada:

**Comment by jywarren** [comment]
Awesome!!

**Pull request merged**